src/cpp/fastdds/domain/DomainParticipantImpl.cpp

```diff
@@ -1745,21 +1741,24 @@ bool DomainParticipantImpl::has_active_entities()
     return false;
 }
 
-void DomainParticipantImpl::set_qos(
+bool DomainParticipantImpl::set_qos(
         DomainParticipantQos& to,
         const DomainParticipantQos& from,
         bool first_time)
 {
+    bool qos_has_changed = false;
```

Comment:
This flag is more "something that should be communicated to the RTPS layer has changed" rather than something has changed, isn't it?

Reply:
I have renamed the flag to `qos_should_be_updated`. It is true that it is not only enough that the QoS has changed because only a few attributes are allowed to be changed.